workshop/content/basics/templates/pseudo-parameters/_index.md

```diff
@@ -0,0 +1,270 @@
+            "Action": [
+                "ssm:GetParameters"
+            ],
+            "Resource": "arn:aws:ssm:us-east-1:111122223333:parameter/dbUsername"
+        }
+    ]
+}
+```
+Let’s see how the CloudFormation template looks like.
+
+First, let’s start by defining the SSM parameter in the template. You will be constructing an ARN for this resource using pseudo parameters. This is a simple example of an SSM parameter where there’s a Name(dbUsername) and Value(alice).
+
+```YAML
+Resources:
+  BasicParameter:
+    Type: AWS::SSM::Parameter
+    Properties:
+      Name: dbUsername
+      Type: String
+      Value: alice
+      Description: SSM Parameter for database username.
+```
+
+Next, let’s define the IAM Role and policy from where you want to reference the SSM parameter you defined above.
+
+```yaml
+  DemoRole:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action:
+              - 'sts:AssumeRole'
+      Path: /
+      Policies:
+        - PolicyName: ssm-least-privilege
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action: 'ssm:GetParameter'
+                Resource: '*'
+  ```
+
+  The IAM role above is a service role for a Lambda function that you plan to deploy, and as it stands, this role will allow a Lambda function to perform GetParameter operation on all SSM parameters. To follow the best practice of least privilege, let’s scope down the IAM policy so that the role lets our Lambda function to only retrieve the SSM Parameter you defined above.
+
+For the DemoRole, under the Policies property, change the Resource: `*` portion to the following:
+
+```yaml
+Resource: !Sub 'arn:${AWS::Partition}:ssm:${AWS::Region}:${AWS::AccountId}:parameter/${BasicParameter}'
+```
+
+In the Resource definition above, instead of using a `*` for Resource, you can construct the ARN of the SSM Parameter you defined above. To construct the ARN for SSM parameter, you use the AWS::Partition, AWS::Region, and AWS::AccountId pseudo parameters. The intrinsic function    **[Fn::Sub](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-sub.html)** lets you substitute variables in an input string with values that you specify. In this case, you can make use of !Sub intrinsic function to substitute the values of the pseudo parameters to the ARN pattern of the SSM parameter.
+
+Finally, add this section below to the template. This will define a Lambda function which uses the IAM Role you defined above with permissions to read the SSM parameter which you also defined above. You will invoke this Lambda function to test if the Lambda function can access the SSM parameter named dbUsername you defined above.
```

Comment:
Instead of using a `*` for Resource in the Resource definition above, you can construct the ARN of the SSM Parameter you defined above. To construct the ARN for SSM parameter, you use the AWS::Partition, AWS::Region, and AWS::AccountId pseudo parameters. The intrinsic function **[Fn::Sub](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-sub.html)** lets you substitute variables in an input string with values that you specify. In this case, you can make use of !Sub intrinsic function to substitute the values of the pseudo parameters to the ARN pattern of the SSM parameter. 	 
  
 
 	 
  
Finally, add this section below to the template. This section defines a Lambda function that uses the IAM Role you defined above with permissions to read the SSM parameter, which you also defined above. You will invoke this Lambda function to test if the Lambda function can access the SSM parameter named dbUsername you defined above.